pkg/cloud/services/elb/loadbalancer.go

```diff
@@ -250,7 +250,7 @@ func (s *Service) getAPIServerLBSpec(elbName string, lbSpec *infrav1.AWSLoadBala
 		// The load balancer APIs require us to only attach one subnet for each AZ.
 		subnets := s.scope.Subnets().FilterPrivate()
 
-		if s.scope.ControlPlaneLoadBalancerScheme() == infrav1.ELBSchemeInternetFacing {
+		if lbSpec.Scheme != nil && *lbSpec.Scheme == infrav1.ELBSchemeInternetFacing {
 			subnets = s.scope.Subnets().FilterPublic()
 		}
```

Comment:
We might want to remove `scope.ControlPlaneLoadBalancerScheme` function altogether, and add defaulting `inline` when that function is used instead?

Reply:
In that function, seems we can get the load balancer spec, and apply the same defaulting logic in place? A local variable should suffice?

Reply:
@vincepri got it. Removed.